rfcs/20180827-api-names.md

```diff
@@ -0,0 +1,593 @@
+| tf.Assert | tf.debugging.Assert |
+| tf.COMPILER_VERSION | tf.version.COMPILER_VERSION |
+| tf.CXX11_ABI_FLAG | tf.sysconfig.CXX11_ABI_FLAG |
+| tf.DType | tf.dtypes.DType |
+| tf.FixedLenFeature | tf.io.FixedLenFeature |
+| tf.FixedLenSequenceFeature | tf.io.FixedLenSequenceFeature |
+| tf.GIT_VERSION | tf.version.GIT_VERSION |
+| tf.GRAPH_DEF_VERSION | tf.version.GRAPH_DEF_VERSION |
+| tf.GRAPH_DEF_VERSION_MIN_CONSUMER | tf.version.GRAPH_DEF_VERSION_MIN_CONSUMER |
+| tf.GRAPH_DEF_VERSION_MIN_PRODUCER | tf.version.GRAPH_DEF_VERSION_MIN_PRODUCER |
+| tf.HistogramProto | tf.summary.HistogramProto |
+| tf.MONOLITHIC_BUILD | tf.sysconfig.MONOLITHIC_BUILD |
+| tf.PaddingFIFOQueue | tf.io.PaddingFIFOQueue |
+| tf.Print | tf.debugging.Print |
+| tf.PriorityQueue | tf.io.PriorityQueue |
+| tf.QUANTIZED_DTYPES | tf.dtypes.QUANTIZED_DTYPES |
+| tf.QueueBase | tf.io.QueueBase |
+| tf.RandomShuffleQueue | tf.io.RandomShuffleQueue |
+| tf.SparseConditionalAccumulator | tf.sparse.SparseConditionalAccumulator |
+| tf.SparseFeature | tf.io.SparseFeature |
+| tf.SparseTensor | tf.sparse.SparseTensor |
+| tf.SparseTensorValue | tf.sparse.SparseTensorValue |
+| tf.SummaryMetadata | tf.summary.SummaryMetadata |
+| tf.VERSION | tf.version.VERSION |
+| tf.VarLenFeature | tf.io.VarLenFeature |
+| tf.abs | tf.math.abs |
+| tf.accumulate_n | tf.math.accumulate_n |
+| tf.add_n | tf.math.add_n |
+| tf.angle | tf.math.angle |
+| tf.argmax | tf.math.argmax |
+| tf.argmin | tf.math.argmin |
+| tf.as_dtype | tf.dtypes.as_dtype |
+| tf.assert_equal | tf.debugging.assert_equal |
+| tf.assert_greater | tf.debugging.assert_greater |
+| tf.assert_greater_equal | tf.debugging.assert_greater_equal |
+| tf.assert_integer | tf.debugging.assert_integer |
+| tf.assert_less | tf.debugging.assert_less |
+| tf.assert_less_equal | tf.debugging.assert_less_equal |
+| tf.assert_near | tf.debugging.assert_near |
+| tf.assert_negative | tf.debugging.assert_negative |
+| tf.assert_non_negative | tf.debugging.assert_non_negative |
+| tf.assert_non_positive | tf.debugging.assert_non_positive |
+| tf.assert_none_equal | tf.debugging.assert_none_equal |
+| tf.assert_positive | tf.debugging.assert_positive |
+| tf.assert_proper_iterable | tf.debugging.assert_proper_iterable |
+| tf.assert_rank | tf.debugging.assert_rank |
+| tf.assert_rank_at_least | tf.debugging.assert_rank_at_least |
+| tf.assert_rank_in | tf.debugging.assert_rank_in |
+| tf.assert_same_float_dtype | tf.debugging.assert_same_float_dtype |
+| tf.assert_scalar | tf.debugging.assert_scalar |
+| tf.assert_type | tf.debugging.assert_type |
+| tf.bfloat16 | tf.dtypes.bfloat16 |
+| tf.bitcast | tf.dtypes.bitcast |
+| tf.bincount | tf.math.bincount |
+| tf.bool | tf.dtypes.bool |
+| tf.cast | tf.dtypes.cast |
+| tf.complex | tf.dtypes.complex |
+| tf.complex128 | tf.dtypes.complex128 |
+| tf.complex64 | tf.dtypes.complex64 |
+| tf.confusion_matrix | tf.train.confusion_matrix |
```

Comment:
Why move `confusion_matrix` under `tf.train.*`? Confusion matrix doesn't have to do with training intrinsically. One can evaluate a model's confusion matrix even without training the model. The `tf.train.*` namespace seems to be mainly about optimizers and checkpoints, etc, which are very different things from confusion matrix.

I think confusion matrix belong more under the `tf.math.*` namespace. Also, `tf.metrics.*` may seem appropriate at first glance, but keep in mind that confusion matrix is not a single scalar, unlike all the things we have under `tf.metrics.* right now.

Reply:
+1

Reply:
Thank you for suggestions! I already made changes in this doc, but we can definitely add more endpoints and deprecate tf.train.confusion_matrix in TensorFlow 2.0.

tf.metrics.* seems to be a good match. Let me confirm if adding tf.metrics.confusion_matrix would be ok.